streaming/base/shared.py

```diff
@@ -1,378 +0,0 @@
-            if os.path.islink(dirname):
-                os.unlink(dirname)
-            shutil.rmtree(dirname, ignore_errors=True)
-
-        atexit.register(cleanup)
-
-    @property
-    def num_enter(self) -> int:
-        """Get property num_enter.
-
-        Returns:
-            int: Number of processes that have entered the barrier.
-        """
-        return self._arr[0]
-
-    @num_enter.setter
-    def num_enter(self, num_enter: int) -> None:
-        """Set property num_enter.
-
-        Args:
-            num_enter (int): Number of processes that have entered the barrier.
-        """
-        self._arr[0] = num_enter
-
-    @property
-    def num_exit(self) -> int:
-        """Get property num_exit.
-
-        Returns:
-            int: Number of processes that have exited the barrier.
-        """
-        return self._arr[1]
-
-    @num_exit.setter
-    def num_exit(self, num_exit: int) -> None:
-        """Set property num_exit.
-
-        Args:
-            num_exit (int): Number of processes that have exited the barrier.
-        """
-        self._arr[1] = num_exit
-
-    @property
-    def flag(self) -> bool:
-        """Get property flag.
-
-        Returns:
-            bool: The flag value.
-        """
-        return bool(self._arr[2])
-
-    @flag.setter
-    def flag(self, flag: bool) -> None:
-        """Set property flag.
-
-        Args:
-            flag (bool): The flag value.
-        """
-        self._arr[2] = bool(flag)
-
-    def __call__(self, num_procs: int) -> None:
-        """A set number of processes enter, wait, and exit the barrier.
-
-        Args:
-            num_procs (int): How many processes are sharing this barrier.
-        """
-        # Re-init the numpy array pointing to shared memory. Necessary when spawn is the
-        # multiprocessing method used.
-        self._arr = np.ndarray(3, buffer=self._shm.buf, dtype=np.int32)
-
-        # Initialize num_exit to the number of processes.
-        with self.lock:
-            if self.num_exit == -1:
-                self.num_exit = num_procs
-
-        # If we are the first to arrive, wait for everyone to exit, then set flag to "don't go".
-        self.lock.acquire()
-        if not self.num_enter:
-            self.lock.release()
-            while self.num_exit != num_procs:
-                sleep(TICK)
-            self.lock.acquire()
-            self.flag = False
-
-        # Note that we entered.
-        self.num_enter += 1
-
-        # If we are the last to arrive, reset `enter` and `exit`, and set flag to "go".
-        if self.num_enter == num_procs:
-            self.num_enter = 0
-            self.num_exit = 0
-            self.flag = True
-        self.lock.release()
-
-        # Everybody waits until the flag is set to "go".
-        while not self.flag:
-            sleep(TICK)
-
-        # Note that we exited.
-        with self.lock:
-            self.num_exit += 1
-
-
-class CreateSharedMemory:
```

Comment:
Why we removed `CreateSharedMemory` ? This class ensures proper cleanup and hides unwanted warning messages.